### PR TITLE
[integration-test] Enable tests to run from local

### PR DESCRIPTION
**Pull request by princerachit**
## Description

Update `TestUploadDownloadBlobViaServer` and `TestUploadDownloadBlob` so that they can run outside gitpod's kubernetes cluster.

### Notable changes
* Determine if the storage provider service is cluster local.
* If it is then create a port forward of the service to localhost.
* Update test methods to add original URL's host as `Host` header. Without this storage provider service such is minio will complain about signature mismatch.
* Rename `ForwardPort` function to `ForwardPortOfPod` and create a new function `ForwardPortOfService`. Both uses common function `forwardPort`

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/9953

## How to test

```console
gitpod /workspace/gitpod/test (princerachit/enhance-the-workspace-9953) $ go test -v ./...   -run "^TestUploadDownloadBlob*"   -kubeconfig /home/gitpod/.kube/config   -namespace default   -username princerachit
?       github.com/gitpod-io/gitpod/test/pkg/agent/daemon       [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/agent/daemon/api   [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/agent/workspace    [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/agent/workspace/api        [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/integration        [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/integration/common [no test files]
=== RUN   TestUploadDownloadBlob
=== RUN   TestUploadDownloadBlob/UploadDownloadBlob
=== RUN   TestUploadDownloadBlob/UploadDownloadBlob/it_should_upload_and_download_blob
    content-service_test.go:199: upload URL: http://localhost:43975/gitpod-user-00000000-0000-0000-0000-000000000000/blobs/test-blob?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=bPggFJB.sPVlgG7oMIWG%2F20220513%2Flocal%2Fs3%2Faws4_request&X-Amz-Date=20220513T114845Z&X-Amz-Expires=1800&X-Amz-SignedHeaders=host&X-Amz-Signature=df246ac5ed45be663e84cecf856069bce1c180e00a9351b70f7bbd7fbe84a20f
    content-service_test.go:212: download URL: http://localhost:41427/gitpod-user-00000000-0000-0000-0000-000000000000/blobs/test-blob?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=bPggFJB.sPVlgG7oMIWG%2F20220513%2Flocal%2Fs3%2Faws4_request&X-Amz-Date=20220513T114847Z&X-Amz-Expires=1800&X-Amz-SignedHeaders=host&X-Amz-Signature=0e08269d26c3f84d6c497aef7d23b131fefc2479401d0f98697c61297e047fa0
--- PASS: TestUploadDownloadBlob (6.70s)
    --- PASS: TestUploadDownloadBlob/UploadDownloadBlob (6.70s)
        --- PASS: TestUploadDownloadBlob/UploadDownloadBlob/it_should_upload_and_download_blob (6.70s)
=== RUN   TestUploadDownloadBlobViaServer
=== RUN   TestUploadDownloadBlobViaServer/UploadDownloadBlobViaServer
=== RUN   TestUploadDownloadBlobViaServer/UploadDownloadBlobViaServer/it_should_uploads_a_blob_via_server_→_content-server_and_downloads_it_afterwards
    content-service_test.go:255: upload URL: http://localhost:40895/gitpod-user-builtin-user-agent-smith-0000000/blobs/test-blob?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=bPggFJB.sPVlgG7oMIWG%2F20220513%2Flocal%2Fs3%2Faws4_request&X-Amz-Date=20220513T114857Z&X-Amz-Expires=1800&X-Amz-SignedHeaders=host&X-Amz-Signature=f379120c7d485d2ed911fc939c423c08faa7de5965047c4fdbe2dc16c6646a46
    content-service_test.go:268: download URL: http://localhost:40985/gitpod-user-builtin-user-agent-smith-0000000/blobs/test-blob?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=bPggFJB.sPVlgG7oMIWG%2F20220513%2Flocal%2Fs3%2Faws4_request&X-Amz-Date=20220513T114859Z&X-Amz-Expires=1800&X-Amz-SignedHeaders=host&X-Amz-Signature=98840aa0471ebb00c81997fc2fa1059c8a0bd0c605fb9c16829a74ba68cac6b9
    content-service_test.go:275: Uploading and downloading blob to content store succeeded.
--- PASS: TestUploadDownloadBlobViaServer (12.67s)
    --- PASS: TestUploadDownloadBlobViaServer/UploadDownloadBlobViaServer (12.67s)
        --- PASS: TestUploadDownloadBlobViaServer/UploadDownloadBlobViaServer/it_should_uploads_a_blob_via_server_→_content-server_and_downloads_it_afterwards (12.67s)
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/content-service       22.256s
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/database      2.846s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/image-builder 2.883s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/server        2.848s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/ws-daemon     2.885s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/ws-manager    2.852s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/ide/jetbrains    2.960s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/ide/vscode       2.901s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/workspace        2.907s [no tests to run]
?       github.com/gitpod-io/gitpod/test/tests/workspace/common [no test files]
```

## Release Notes

```release-note
None
```

### CODE REVIEW

**Comment by roboquat** [comment]
@princerachit: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: test/pkg/integration/apis.go [hunk 1]
```diff
@@ -157,6 +158,39 @@ func EncryptValue(value []byte, key []byte) (data string, iv string) {
...
+// it takes a url as input and creates a port forward if required
+// e.g. when minio running in gitpod cluster
+// and modifies the url to refer to the localhost instead of dns name
+func (c *ComponentAPI) Storage(connUrl string) (string, error) {
+	u, err := url.Parse(connUrl)
+	if err != nil {
+		return "", err
+	}
+	host, port, _ := net.SplitHostPort(u.Host)
+	if !strings.HasSuffix(host, ".svc.cluster.local") {
```
Comment (princerachit) [inline test/pkg/integration/apis.go:171]: Check if the storage provider service is running in the cluster

File: test/pkg/integration/common/port_forward.go [hunk 2]
```diff
@@ -18,15 +18,23 @@ import (
 // ForwardPort establishes a TCP port forwarding to a Kubernetes pod
```
Comment (jenting) [inline test/pkg/integration/common/port_forward.go:18]: The function name requires to change as well.

**Comment by princerachit** [comment]
> Question: could we port forward MinIO pod instead of port forward MinIO service?

Yes we can. The reason I used service instead of a pod is because the URL configured for content service points to minio service and the DNS can be easily used to translate to corresponding service. In our case minio has only one replica but it can have several replicas if configured as a distributed stateful set.

I am open to any feedback if you think this is not a good idea.

**Review by jenting — APPROVED** [review]
/lgtm

File: test/tests/components/content-service/content-service_test.go [hunk 3]
```diff
@@ -263,12 +281,20 @@ func TestUploadDownloadBlobViaServer(t *testing.T) {
...
+		t.Fatal(err)
+	}
 	var client = &http.Client{Timeout: time.Second * 10}
-	httpreq, err := http.NewRequest(http.MethodPut, url, strings.NewReader(content))
+	httpreq, err := http.NewRequest(http.MethodPut, updatedUrl, strings.NewReader(content))
 	if err != nil {
 		t.Fatalf("cannot create HTTP PUT request: %q", err)
 	}
+	// Add Host header
+	httpreq.Host = u.Host
```
Comment (jenting) [inline test/tests/components/content-service/content-service_test.go:297]: ❤️

**Pull request merged**